dynamic_programming/range_maximum_query.py

```diff
@@ -0,0 +1,133 @@
+"""
+Range Maximum Query Problem:
+
+    Given an array of integers and q queries,
+    For each query find the maximum value in the range [l, r).
+
+    The array is 0-indexed and the queries are 0-indexed.
+
+    More details:
+        https://www.geeksforgeeks.org/range-maximum-query-using-sparse-table/
+"""
+from __future__ import annotations
+
+import typing
+
+
+class RangeMaximumQuery:
+    def __init__(self, array: typing.Sequence[int | float]) -> None:
+        """
+        Initialize RangeMaximumQuery with given array.
+
+        Example:
+        >>> rmq = RangeMaximumQuery([1, 3, 5, 7, 9, 11])
+        """
+        self.array = array
+        self._preprocess()
```

Comment:
The names `log` (for `logarithm_array`) and `dp` (for `dynamic_programming` ?!?) are not self-documenting.

The principle of least surprise says that `self.log` and `self.dp` should be defined directly in `.__init__()` so:
* [ ] Remove `_preprocess()` and put its contents into `.__init__()`
* [ ] Move the contents of `_preprocess()` into `make_log()` and `make_dp()` methods that are called in `__init__()`
* [ ] Set `self.log = None` and `self.dp = None` in `.__init__()` -- (My least favorite solution)

Reply:
Is it good now?